code/modules/fissionreactor/fission_datums.dm

```diff
@@ -0,0 +1,668 @@
+/*
+IN THIS FILE:
+datums for the fission reactor, which includes the fuel and reactor
+*/
+#define FISSIONREACTOR_MELTDOWNTEMP 5500 //temp when shit goes wrong
+#define FISSIONREACTOR_DANGERTEMP 4500 //temp to start warning you and to SCRAM
+#define FISSIONREACTOR_SAFEENUFFTEMP 1000 //temp where SCRAM resets
+
+/datum/fission_reactor_holder
+	var/list/fuel_rods=list() //phase 0 vars, set upon construction
+	var/list/control_rods=list()
+	var/list/coolant_ports=list()
+	var/list/casing_parts=list()
+	var/list/breaches=list()
+	var/obj/machinery/fissioncontroller/controller=null
+	var/obj/ticker=null //what ticks it.
+	var/heat_capacity=0
+	var/fuel_reactivity=1
+	var/fuel_reactivity_with_rods=0
+	var/fuel_rods_affected_by_rods=0
+	
+	var/coolantport_counter=0 // this varible exists to ensure that all coolant ports get treated equally, because if we didn't it would have a flow prefrence towards the ports with lower indexes.
+	var/control_rod_insertion=1  //phase 1 vars. modified during runtime
+	var/control_rod_target=1 // this is to create a bit of input lag to make things a bit more tense. also allows autoscram to work while the controller is unpowered.
+	var/SCRAM=FALSE //all caps because AAAAAAAAAAAAAAAAAAA EVERYBODY PANIC WE'RE ALL GONNA DIE.
+	var/temperature=T20C //this is set last
+	
+	var/datum/gas_mixture/coolant
+
+	var/graceperiodtick=2 // set to true when we hit meltdown temp. gives you a bit of time to GTFO or save it. this will result in peak kino (i hope)
+	
+	var/zlevel=0 //positional varibles
+	var/origin_x=0
+	var/origin_y=0
+	var/corner_x=0 //uses corner calculations. this is for the sake of being easier to calculate.
+	var/corner_y=0
+	var/datum/fission_fuel/fuel=null
+
+	
+/datum/fission_reactor_holder/New()
+	..()
+	coolant = new /datum/gas_mixture
+	coolant.temperature = T20C //vaguely room temp.
+	coolant.volume = CELL_VOLUME
+	fissionreactorlist+=src
+
+/datum/fission_reactor_holder/Destroy()
```

Comment:
Per the lint, Destroy() **must** call its parent!

Reply:
fixed in upcoming commit